src/layer/dataset.ts

```diff
@@ -1,10 +1,27 @@
+enum ShDatasetParametersType {
+  BYOC = 'BYOC',
+  S1GRD = 'S1GRD',
+  S2 = 'S2',
+  S2GM = 'S2GM',
+  S3 = 'S3',
+  S3SLSTR = 'S3SLSTR',
+  S5PL2 = 'S5PL2',
+}
+
+enum EocDatasetParametersType {
+  S1GRD = 'S1GRD',
+  S5PL2 = 'S5PL2',
+}
```

Comment:
I guess that's more of a personal preference, but I find it more readable if strings are used instead of `enum`s in this case. It is also more consistent with how other values are defined in `Dataset`. Do you think we can use the strings directly?

Reply:
Made a suggested change.